sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -66,9 +66,21 @@ private[sql] class DefaultSource
       mode match {
         case SaveMode.Append =>
           sys.error(s"Append mode is not supported by ${this.getClass.getCanonicalName}")
-        case SaveMode.Overwrite =>
-          fs.delete(filesystemPath, true)
+        case SaveMode.Overwrite => {
+          try {
+            if (!fs.delete(filesystemPath, true)) {
+              throw new IOException(
+                s"Unable to clear output directory ${filesystemPath.toString} prior"
+                  + s" to INSERT OVERWRITE a JSON table:\n")
+            }
+          } catch {
+            case e: IOException =>
+              throw new IOException(
+                s"Unable to clear output directory ${filesystemPath.toString} prior"
+                  + s" to INSERT OVERWRITE a JSON table:\n${e.toString}")
```

Comment:
This will also catch the one thrown by `try`. Can we avoid this? If `delete` returns false, we just throw the exception after the `catch`. Also, you can put `e` as the cause of the new exception, like `IOException("...", e)`.